### ISSUE TITLE
Transfer songs to a device with file organisation and playlists

### ISSUE THREAD

**Issue by Isaacson33** [jonaski/strawberry#252]
It would be great if files could be transferred to a device by playlist with all the file organisation intact.

So, instead of moving the files to the main directory, files would be copied with their file structure, so something on the PC in /Music/Artist/Album/song.mp3 would appear on the device under /Music/Artist/Album/song.mp3. That way, one could make a playlist on Strawberry and transfer the playlist to an external device, together with its m3u file which would faithfully point to all the right directory entries for all the songs even on the external device.

**Comment by jonaski** [comment]
It's already done for mounted file system devices.
For gpod devices, it uses special iTunes type hierarchy 
For MTP devices, it's not possible as for as I can see, according to the libmtp documentation, you only specify the source file, not the destination folder.
http://blackjack.izmiran.ru/doc/libmtp-0.0.18/html/group__tracks.html#ga1f986d6a026a641810e56c5dc63003a

**Comment by zsrv** [comment]
Is it possible for this to be revisited? In the Doxygen documentation for libmtp 1.1.20 (I couldn't find a web link to the documentation but I've attached the full example), sendtr.c in the examples section was updated to allow the remote path to be specified for the track to be sent.

The problem I ran into: Since all music files are currently copied into the same directory on the MTP device, I see copy failures if music files from different directories on my computer happen to have the same file name.

[sendtr.c.txt](https://github.com/strawberrymusicplayer/strawberry/files/10098489/sendtr.c.txt)
